WcaOnRails/app/controllers/admin_controller.rb

```diff
@@ -26,4 +26,43 @@ def do_merge_people
     end
     render 'merge_people'
   end
+
+  def edit_person
+    # This is necessary because the simple form needs some real active record when using user_ids picker.
+    # That's only to pass the appropriate attributes.
+    @person = Person.new
+  end
+
+  def update_person
+    @person = Person.current.find_by!(wca_id: params[:person][:wca_id])
+    person_params = params.require(:person).permit(:name, :countryId, :gender, :dob)
+
+    case params[:commit]
+    when "Fix"
+      if @person.update_attributes(person_params)
+        flash.now[:success] = "Successfully fixed #{@person.name}."
+        if @person.country_id_changed
+          flash.now[:warning] = "The change you made may have affected national and continental records, be sure to run
+          <a href='/results/admin/check_regional_record_markers.php'>check_regional_record_markers</a>.".html_safe
+        end
+      end
+    when "Update"
+      if @person.update_using_sub_id(person_params)
+        flash.now[:success] = "Successfully updated #{@person.name}."
```

Comment:
Same comment as before: the Rails convention is to redirect after a successful database operation.